src/libopensc/card-iasecc.c

```diff
@@ -1544,6 +1544,7 @@ iasecc_delete_file(struct sc_card *card, const struct sc_path *path)
 	struct sc_apdu apdu;
 	struct sc_file *file = NULL;
 	int rv;
+	unsigned int file_id;
```

Comment:
`struct sc_file` member `id` is defined as a signed int so I think we should keep that (technically it should not matter as it should be just 2 bytes, but some future compiler might start complaining):
https://github.com/OpenSC/OpenSC/blob/c2860b0501a2f44307a063d4e4886ea37ec8306e/src/libopensc/types.h#L263

Reply:
In that case, we have a problem in the following function prototype (where `file_id` is used):
`````
src/libopensc/iasecc-sm.c:iasecc_sm_delete_file(struct sc_card *card, unsigned se_num, unsigned int file_id)
src/libopensc/iasecc-sdo.h:int iasecc_sm_delete_file(struct sc_card *card, unsigned se_num, unsigned int file_id);
`````
We even find another conversion there:
`sm_info->cmd_data = (void *)(uintptr_t)file_id;`

Should I fix it everywhere?

Reply:
Oh. If it would be just this one internal function, I would go ahead and fix it to avoid needless conversion.
The other cast is weird. The cast back is in `src/smm/sm-card-iasecc.c`, but only for logging if I see right and I do not see if it is used for something else deeper in the code so I would probably not touch it and keep it as it is.

Reply:
At this point, it doesn't really matter if I use `unsigned int file_id;` or `int file_id;` because the conversion will happen either way and this conversion is completely unambiguous in the case of a 16-bit number.

I think we should leave this PR as it is. Any fix  related to `file_id` (unsigned/signed) will take place in a separate PR .. and also a separate PR will be used for the problem with `card->cache.valid` and `card->cache.current_ef`.

Reply:
Agreed. Thanks for looking into this comment!